src/storage/src/hummock/shared_buffer/shared_buffer_manager.rs

```diff
@@ -68,20 +121,74 @@ impl SharedBufferManager {
             shared_buffer: PLRwLock::new(BTreeMap::new()),
             uploader_tx,
             uploader_handle,
+            stats,
+            ongoing_flush: OnGoingFlush::new(),
         }
     }
 
+    fn empty(&self) -> bool {
+        self.stats.shared_buffer_cur_size.load(Ordering::SeqCst) == 0
+    }
+
+    /// Allocates a shared buffer budget.
+    async fn allocate_space(&self, batch_size: u64) -> HummockResult<()> {
+        let mut current_size = self.stats.shared_buffer_cur_size.load(Ordering::SeqCst);
+        let threshold = self.stats.shared_buffer_threshold_size;
+
+        // Atomically allocates space,
+        // since there could be concurrent Actors write to the shared buffer.
+        'retry_allocate: loop {
+            // flush shared buffer if there is no enough space
+            while threshold < current_size + batch_size {
+                log::debug!(
+                    "trigge flush: threshold {}, new_size {}",
+                    threshold,
+                    current_size + batch_size
+                );
+                self.flush().await?;
+                current_size = self.stats.shared_buffer_cur_size.load(Ordering::SeqCst);
+            }
```

Comment:
If the batch size is larger than the threshold (256 MB by default), the loop will never end...

Reply:
If that's the case. it will fail at the `assert(!self.empty())` in `flush()` now.